frontend/src/views/ProblemPage.tsx

```diff
@@ -23,68 +22,29 @@ type ProblemParams = {
   id: string,
 };
 
-type LocationState = {
-  locked: boolean,
-};
-
 function ProblemPage() {
-  const history = useHistory();
-  const location = useLocation<LocationState>();
+  const { firebaseUser, token } = useAppSelector((state) => state.account);
+
   const [problem, setProblem] = useState<Problem | null>(null);
   const [error, setError] = useState('');
   const [loading, setLoading] = useState(false);
 
+  const problemEditable = useProblemEditable(firebaseUser, problem);
   const params = useParams<ProblemParams>();
 
-  // The problems page is loading or locked until a valid password is supplied.
-  const [locked, setLocked] = useState<boolean | null>(null);
-
-  useEffect(() => {
-    if (checkLocationState(location, 'locked')) {
-      setLocked(location.state.locked);
-    } else {
-      setLocked(true);
-    }
-  }, [location]);
-
   useEffect(() => {
-    if (!locked) {
-      setLoading(true);
-      getSingleProblem(params.id)
-        .then((res) => {
-          res.testCases.forEach((testCase) => {
-            // eslint-disable-next-line no-param-reassign
-            testCase.id = generateRandomId();
-          });
-          setProblem(res);
-          setLoading(false);
-        })
-        .catch((err) => {
-          setError(err.message);
-          setLoading(false);
+    setLoading(true);
+    getSingleProblem(params.id)
+      .then((res) => {
+        res.testCases.forEach((testCase) => {
+          // eslint-disable-next-line no-param-reassign
```

Comment:
nit: We may not need this anymore given that this ESLint warning is disabled site-wide? (If this is true, it may be worth removing all instances of that comment throughout the codebase, if we think that site-wide warning disabling will remain.)

Reply:
Hm I'm still getting a warning:

![no-param-reassign warning in IntelliJ](https://user-images.githubusercontent.com/24768574/119211760-dfdb7a00-ba68-11eb-9411-baab7039c8f2.png)

I think it's because the ESLint rule is only for variables named `state` (specifically to allow param reassign with Redux) but remains the same for all other variables. Thoughts @zpChris? Could either leave this as is or loosen that ESLint requirement to all variables

Reply:
Nah don't make any changes, just leaving that one comment there works fine.